### PR TITLE
Feature/multithreaded public key

### PR DESCRIPTION
**Pull request by aPere3**
### Resolves:

closes https://github.com/zama-ai/concrete-core-internal/issues/409
closes https://github.com/zama-ai/concrete-core-internal/issues/268

### Description

This PR adds multithreaded public key generation to the api.

### Checklist 

(Use '[x]' to check the checkboxes, or submit the PR and then click the checkboxes)

* [x] Tests for the changes have been added (for bug fixes / features)
* [x] Docs have been added / updated (for bug fixes / features)
* [x] The PR description links to the related issue (to link an issue, use '#XXX'.)
* [ ] The tests on AWS have been launched and are successful (comment with @slab-ci cpu_test and/or @slab-ci gpu_test to trigger the tests)
* [ ] The draft release description has been updated
* [ ] Check for breaking changes (including serialization changes) and add them to commit message following the conventional commit [specification][conventional-breaking]

[conventional-breaking]: https://www.conventionalcommits.org/en/v1.0.0/#commit-message-with-description-and-breaking-change-footer

### CODE REVIEW

**Comment by IceTDrinker** [comment]
should we convert this to a proper PR ?

**Comment by aPere3** [comment]
@slab-ci cpu_test

**Comment by agnesLeroy** [comment]
@aPere3 the tests fail it seem?

**Comment by aPere3** [comment]
Apparently yes. Let me have a look !

**Comment by aPere3** [comment]
@slab-ci cpu_test

File: concrete-core-test/src/default.rs [hunk 1]
```diff
@@ -165,3 +165,38 @@ test! {
...
+                test_parallel!{($($key_dist),*), $fixture, Precision64, ($([< $types 64 >]),+)}
+            }
+        )+
+    };
+}
+
+#[cfg(feature = "backend_default_parallel")]
+test_parallel! {
+    ((BinaryKeyDistribution), LweCiphertextVectorZeroEncryptionFixture, (LweSecretKey, LweCiphertextVector)),
+    ((BinaryKeyDistribution), LwePublicKeyGenerationFixture, (LweSecretKey, LwePublicKey))
```
Comment (agnesLeroy) [inline concrete-core-test/src/default.rs:201]: Is it normal that this is still here? I thought you had removed it?

**Review by agnesLeroy — COMMENTED** [review]
Looks good to me! I just thought you had removed the test for public key generation but it seems to still be here?

File: concrete-core-test/src/default.rs (hunk 1 above)
Comment (aPere3) [reply concrete-core-test/src/default.rs:201]: Forgot that one ! Will do.

**Comment by aPere3** [comment]
@slab-ci cpu_test

**Review by agnesLeroy — APPROVED** [review]

**Pull request merged**